File: Formula-Bollo-App/src/main/java/formula/bollo/app/entity/Result.java
package formula.bollo.app.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;

import lombok.Data;

@Entity
@Table(name = "result")
@Data
public class Result {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @ManyToOne
    @JoinColumn(name = "race_id")
    private Race race;

    @ManyToOne
    @JoinColumn(name = "driver_id")
    private Driver driver;

    @ManyToOne
    @JoinColumn(name = "position_id")
    private Position position;

    @Column(name = "fastlap", nullable = false, length = 1)
    private int fastlap;

    @ManyToOne
    @JoinColumn(name = "season_id")
    private Season season;

    @Column(name = "pole", nullable =  false, length = 1)
    private int pole;
}

File: Formula-Bollo/src/shared/models/result.ts
import { Driver } from "./driver";
import { Position } from "./position";
import { Race } from "./race";
import { Season } from "./season";

export class Result {
  id: number;
  race: Race | null;
  driver: Driver;
  position: Position | null;
  fastlap: number;
  pole: number;
  season: Season | undefined;

  constructor(
    id: number,
    race: Race | null,
    driver: Driver,
    position: Position | null,
    fastlap: number,
    pole: number,
    season?: Season,
  ) {
    this.id = id;
    this.race = race;
    this.driver = driver;
    this.position = position;
    this.fastlap = fastlap;
    this.pole = pole;
    this.season = season;
  }
}
